tests/network/vmi_istio.go

```diff
@@ -0,0 +1,207 @@
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/libnet"
+	"kubevirt.io/kubevirt/tests/libvmi"
+)
+
+const (
+	istioDeployedEnvVariable     = "KUBEVIRT_DEPLOY_ISTIO"
+	istioInjectSidecarAnnotation = "sidecar.istio.io/inject"
+	istioInjectNamespaceLabel    = "istio-injection"
+	vmiDefaultPortsAppSelector   = "istio-vmi-default-ports"
+	vmiExplicitPortsAppSelector  = "istio-vmi-explicit-ports"
+	vmiDefaultPortsAppCaseName   = "vmiWithDefaultPorts"
+	vmiExplicitPortsAppCaseName  = "vmiWithExplicitPorts"
+	svcDeclaredTestPort          = 1500
+	svcUndeclaredTestPort        = 1501
+)
+
+type vmiAppCase struct {
+	appSelector string
+	vmi         *v1.VirtualMachineInstance
+	ports       []v1.Port
+	service     *corev1.Service
+}
+
+func istioServiceMeshDeployed() bool {
+	value := os.Getenv(istioDeployedEnvVariable)
+	if value == "true" {
+		return true
+	}
+	return false
+}
+
+func withAppLabel(value string) libvmi.Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		if vmi.Labels == nil {
+			vmi.Labels = map[string]string{}
+		}
+		vmi.Labels["app"] = value
+	}
+}
+
+func withIstioAnnotation() libvmi.Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		if vmi.Annotations == nil {
+			vmi.Annotations = map[string]string{}
+		}
+		vmi.Annotations[istioInjectSidecarAnnotation] = "true"
+	}
+}
+
+func newServiceWithAppSelector(name, namespace, selector string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"app": selector,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Port: svcDeclaredTestPort,
+				},
+			},
+		},
+	}
+}
+
+func newRandomCirrosVMIWithIstioSidecarAndDefaultServiceAccountAndNetworkAndInterface(appLabel string, network *v1.Network, ports []v1.Port) *v1.VirtualMachineInstance {
+	vmi := libvmi.NewCirros(
+		libvmi.WithNamespace(tests.NamespaceTestDefault),
+		libvmi.WithNetwork(network),
+		libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding(ports...)),
+		withIstioAnnotation(),
+		withAppLabel(appLabel),
+	)
+	tests.AddServiceAccountDisk(vmi, "default")
+	return vmi
+}
+
+var _ = SIGDescribe("[Serial] Istio", func() {
+	var (
+		err        error
+		virtClient kubecli.KubevirtClient
+	)
+	network := &v1.Network{
+		Name: "default",
+		NetworkSource: v1.NetworkSource{
+			Pod: &v1.PodNetwork{},
+		},
+	}
+	explicitPorts := []v1.Port{
+		{Port: svcDeclaredTestPort},
+		{Port: svcUndeclaredTestPort},
+	}
+	vmiAppCases := map[string]*vmiAppCase{
+		vmiDefaultPortsAppCaseName: {
+			appSelector: vmiDefaultPortsAppSelector,
+			ports:       []v1.Port{},
+		},
+		vmiExplicitPortsAppCaseName: {
+			appSelector: vmiExplicitPortsAppSelector,
+			ports:       explicitPorts,
+		},
+	}
+
+	Describe("Virtual machine in a service mesh using masquerade interface with explicit port list", func() {
+		tests.BeforeAll(func() {
```

Comment:
We are trying to avoid the usage of `BeforeAll` in tests that are [not serial](https://github.com/kubevirt/kubevirt/pull/4051).
`BeforeAll` means that if each test will run on a different process the `BeforeAll` will be invoked as the number of the processes, which may be even less efficient than just having `BeforeEach` for every relevant test.

Reply:
Refactored the test suite - I ditched BeforeAll in favor of BeforeEach and AfterEach.